frontend/src/components/UserProfile/UserName.tsx

```diff
@@ -0,0 +1,180 @@
+// Copyright 2023 The Matrix.org Foundation C.I.C.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {
+  Alert,
+  Control,
+  Field,
+  Label,
+  Root,
+  Button,
+} from "@vector-im/compound-web";
+import { useAtomValue, useAtom, useSetAtom } from "jotai";
+import { atomFamily } from "jotai/utils";
+import { atomWithQuery, atomWithMutation } from "jotai-urql";
+import {
+  useState,
+  useEffect,
+  ChangeEventHandler,
+  FormEventHandler,
+} from "react";
+
+import { graphql } from "../../gql";
+import LoadingSpinner from "../LoadingSpinner/LoadingSpinner";
+
+import styles from "./UserName.module.css";
+
+const QUERY = graphql(/* GraphQL */ `
+  query UserGreeting($userId: ID!) {
+    user(id: $userId) {
+      id
+      username
+      matrix {
+        mxid
+        displayName
+      }
+    }
+  }
+`);
+
+const userGreetingFamily = atomFamily((userId: string) => {
+  const userGreeting = atomWithQuery({
+    query: QUERY,
+    getVariables: () => ({ userId }),
+  });
+
+  return userGreeting;
+});
```

Comment:
You should instead export the `userGreeting` atom family from the `UserGreeting` component & reimport it here